src/util/psbt/mod.rs

```diff
@@ -91,6 +91,29 @@ impl PartiallySignedTransaction {
     }
 }
 
+#[cfg(feature = "base64")]
+mod _base64encoding {
+    use super::{PartiallySignedTransaction, Error};
+    use std::fmt::{Display, Formatter, self};
+    use std::str::FromStr;
+
+    impl Display for PartiallySignedTransaction {
+        fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+            use consensus::encode::serialize;
+            f.write_str(&::base64::encode(&serialize(self)))
+        }
+    }
+
+    impl FromStr for PartiallySignedTransaction {
+        type Err = Error;
+
+        fn from_str(s: &str) -> Result<Self, Self::Err> {
+            use consensus::encode::deserialize;
+            Ok(deserialize(&::base64::decode(s).map_err(|_| Error::ConsensusEncoding)?)?)
```

Comment:
Swallowing the underlying error is suboptimal since it's unclear if it came from the base64 or from parsing the binary data structure (unless it is only used here which I doubt given the description and pre-existance).

Reply:
addressed in the commit comment of a6399e7